### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by OleksandrKukla**

### CODE REVIEW

File: src/components/BackgroundContainer.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { string } from 'prop-types';
+
+BackgroundContainer.propTypes = {
+    img: string
+};
```
Comment (xxxMortxxx) [inline src/components/BackgroundContainer.js]: Add types for children and filter properties

File: src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+export default function ({ title, size, addClasses, type, isActive }) {
+    let typeClasses = {
```
Comment (xxxMortxxx) [inline src/components/Button.js]: Looks like a constant. You can move this code out of the function:
```
let typeClasses = {
            outline: 'btn-outline-primary',
            secondary: 'btn-secondary'
        },
        sizeClasses = {
            small: 'btn-sm'
      };
```

File: src/pages/Details.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    updateCurrentMovie () {
+        this.currentMovie = mockObject.find((el) => (
+            String(el.id) === this.props.match.params.movieID
+        ));
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <React.Fragment>
```
Comment (xxxMortxxx) [inline src/pages/Details.js]: Looks like one of this <React.Fragment> is redundant

File: src/pages/Details.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import HelpLine from '../components/HelpLine';
+import MovieList from '../components/MovieList';
+
+import mockObject from '../MockFilmList';
+
+export default class Details extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.updateCurrentMovie();
```
Comment (xxxMortxxx) [inline src/pages/Details.js]: To store internal component data which affects view, use state object.

```
constructor(props) {
  super(props);
  this.state = {
    currentMovie: mockObject.find((el) => el.id === props.match.params.movieID)
  }
}
```

File: src/components/Search.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+import Button from './Button';
+
+export default class Search extends React.PureComponent {
+    constructor(props) {
```
Comment (xxxMortxxx) [inline src/components/Search.js]: If you're not setting the state in the constructor, or not doing some other logic, there is no need to write a constructor function. 'super' will be called automatically

File: src/components/MovieList.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+import MovieListItem from './MovieListItem';
+
+export default class MovieList extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.isEmpty = !(this.props.movieList && this.props.movieList.length);
```
Comment (xxxMortxxx) [inline src/components/MovieList.js]: Use state to store local component data which affects view.

```
constructor(props) {
        super(props);
        this.isEmpty = !(this.props.movieList && this.props.movieList.length);
        this.state = {
    isEmpty: !(this.props.movieList && this.props.movieList.length),
  }
}
```

Comment (OleksandrKukla) [reply src/components/MovieList.js]: Done

File: src/components/Search.js (hunk 5 above)
Comment (OleksandrKukla) [reply src/components/Search.js]: Done

File: src/pages/Details.js (hunk 4 above)
Comment (OleksandrKukla) [reply src/pages/Details.js]: Done

File: src/pages/Details.js (hunk 3 above)
Comment (OleksandrKukla) [reply src/pages/Details.js]: Done

File: src/components/Button.js (hunk 2 above)
Comment (OleksandrKukla) [reply src/components/Button.js]: Done

File: src/components/BackgroundContainer.js (hunk 1 above)
Comment (OleksandrKukla) [reply src/components/BackgroundContainer.js]: Done

**Pull request merged**